Sources/StreamChatUI/ChatMessageList/ChatMessageListVC.swift

```diff
@@ -449,13 +419,31 @@ open class ChatMessageListVC: _ViewController,
     }
 
     open func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+        if let message = dataSource?.chatMessageListVC(self, messageAt: indexPath) {
+            cellHeightsCache[message.id] = cell.bounds.size.height
+        }
+
         delegate?.chatMessageListVC(self, willDisplayMessageAt: indexPath)
     }
 
+    open func tableView(_ tableView: UITableView, estimatedHeightForRowAt indexPath: IndexPath) -> CGFloat {
+        if let message = dataSource?.chatMessageListVC(self, messageAt: indexPath) {
+            return cellHeightsCache[message.id] ?? UITableView.automaticDimension
+        }
+
+        return UITableView.automaticDimension
+    }
+
     open func scrollViewDidScroll(_ scrollView: UIScrollView) {
         delegate?.chatMessageListVC(self, scrollViewDidScroll: scrollView)
 
         setScrollToLatestMessageButton(visible: isScrollToBottomButtonVisible)
+
+        // If the user scrolled to the bottom, update the UI for the skipped messages
+        if listView.isLastCellFullyVisible && !listView.skippedMessages.isEmpty {
```

Comment:
I am a bit concerned about the pieces of custom logic. Is this something we can solve any other way? This can be one of those things that we end up maintaining without knowing how to remove

Reply:
Unfortunately, we don't have any other option. Of all the solutions I tried, none fixes the jumps in the message list. We have been dealing with jumps in the message list since I joined Stream 😅 When I thought about this solution I was thinking it could be problematic and error-prone, but actually after talking with @martinmitrevski, this is what they are actually doing on SwiftUI. So if it is working for them, and it is stable, it means we can do it as well.

Reply:
yes, this works well in SwiftUI, so far we haven't received any complaints. However, it's important that we find a way to apply any other updates (e.g. reactions, deletes), since this is the part where it differs from SwiftUI atm.